src/NuGet.Core/NuGet.ProjectModel/StringExtensions.cs

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+#nullable enable
+
+using System;
+
+namespace NuGet.ProjectModel
+{
+    internal static class StringExtensions
+    {
+        internal static (string firstPart, string? secondPart) SplitInTwo(this string s, char separator)
+        {
+            if (string.IsNullOrEmpty(s))
+            {
+                return (s, null);
+            }
+#if NETCOREAPP2_1_OR_GREATER
+            var index = s.IndexOf(separator, StringComparison.OrdinalIgnoreCase);
+#else
+            var index = s.IndexOf(separator.ToString(), StringComparison.OrdinalIgnoreCase);
```

Comment:
Instead of this, call [this overload](https://learn.microsoft.com/en-us/dotnet/api/system.string.indexof?view=net-8.0#system-string-indexof(system-char), which defaults to 'Ordinal'. You only ever pass a slash to this method so the results will not change.